Core/src/org/sleuthkit/autopsy/datamodel/BaseChildFactory.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.datamodel;
+
+import com.google.common.collect.Lists;
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.openide.nodes.ChildFactory;
+import org.sleuthkit.autopsy.core.UserPreferences;
+import org.sleuthkit.datamodel.Content;
+
+/**
+ * Abstract child factory that provides paging and filtering functionality to
+ * subclasses.
+ *
+ * @param <T>
+ */
+public abstract class BaseChildFactory<T extends Content> extends ChildFactory.Detachable<T> {
+
+    private final Predicate<T> filter;
+    private boolean isPageChangeEvent;
+
+    private final PagingSupport pagingSupport;
+
+    /**
+     * This static map is used to facilitate communication between the UI and
+     * the child factory.
+     */
+    public static Map<String, EventBus> nodeNameToEventBusMap = new ConcurrentHashMap<>();
```

Comment:
I don't like the idea of a global variable here. Would it make sense to have static register, post, etc. methods that take a node name and delegate to the corresponding method on the fully encapsulated bus objects? Also, I am also concerned about the way update and retrieval operations can overlap in ConcurrentHashMap.

Reply:
I like the idea of static methods and will look into making that change.
I don't understand your concerns about map update and retrieval. Entries are added to the map once (and never updated) when the child factory is created and they are retrieved when publishers and subscribers need them. We should probably talk about this.

Reply:
We are going with the static methods, then, coming in a later commit. I am going to take this now.